rest-api/test/unit_test/dao_test/biobank_order_dao_test.py

```diff
@@ -118,3 +172,139 @@ def test_store_invalid_test(self):
       self.dao.insert(self._make_biobank_order(
           samples=[BiobankOrderedSample(
               test='InvalidTestName', processingRequired=True, description=u'tested it')]))
+
+  def test_cancelling_an_order(self):
+    ParticipantSummaryDao().insert(self.participant_summary(self.participant))
+    order_1 = self.dao.insert(self._make_biobank_order())
+    cancelled_request = self._get_cancel_patch()
+    updated_order = self.dao.update_with_patch(order_1.biobankOrderId, cancelled_request,
+                                               order_1.version)
+
+    self.assertEqual(updated_order.version, 2)
+    self.assertEqual(updated_order.cancelledUsername, 'mike@pmi-ops.org')
+    self.assertEqual(updated_order.orderStatus, BiobankOrderStatus.CANCELLED)
+    self.assertEqual(updated_order.amendedReason, cancelled_request['amendedReason'])
+
+  def test_cancelled_order_removes_from_participant_summary(self):
+    ParticipantSummaryDao().insert(self.participant_summary(self.participant))
+    samples = [BiobankOrderedSample(
+      test=self._B_TEST, processingRequired=True, description=u'new sample')]
+    biobank_order_id = 2
+    order_1 = self.dao.insert(self._make_biobank_order())
+    self.dao.insert(self._make_biobank_order(samples=samples,
+                                             biobankOrderId=biobank_order_id,
+                                             identifiers=[
+                                                         BiobankOrderIdentifier(system='z',
+                                                                                value='x')]))
+    cancelled_request = self._get_cancel_patch()
+    ps_dao = ParticipantSummaryDao().get(self.participant.participantId)
+
+    self.assertEqual(ps_dao.sampleOrderStatus1ED10, OrderStatus.CREATED)
+    self.assertEqual(ps_dao.sampleOrderStatus2ED10, OrderStatus.CREATED)
+    self.dao.update_with_patch(order_1.biobankOrderId, cancelled_request,
+                               order_1.version)
+    ps_dao = ParticipantSummaryDao().get(self.participant.participantId)
+
+    self.assertEqual(ps_dao.sampleOrderStatus1ED10, None)
```

Comment:
Check the timestamp values, too?

You can do:

 with FakeClock(TIME_1):
  <insert order 1>
with FakeClock(TIME_2):
  <insert order 2>

to make sure the timestamps differ.

Reply:
done